docs/ref/applications.txt

```diff
@@ -509,3 +514,26 @@ Here are some common problems that you may encounter during initialization:
   :setting:`INSTALLED_APPS` to contain
   ``'django.contrib.admin.apps.SimpleAdminConfig'`` instead of
   ``'django.contrib.admin'``.
+
+* ``RuntimeWarning: Accessing the database during app initialization is
+  discouraged.`` This happens if one of your apps attempts database access in
+  either the :meth:`AppConfig.ready` method or during import time. This is bad
+  because it will happen during the startup of every management command.
+
+  For example, a common mistake may be to populate the choices of a form field
+  like this::
+
+    class SomeForm(forms.Form):
+        country = forms.ChoiceField(choices=[c.name for c in Country.objects.all()])
+
+  In the example above, the ``Country.objects.all()`` query would be executed
+  as soon as the module containing this form would be imported. To silence the
+  warning, use a :class:`~django.forms.ModelChoiceField` instead::
+
+    class SomeForm(forms.Form):
+        country = forms.ModelChoiceField(queryset=Country.objects.all())
+
+  To make it easier to find the code that triggered this warning, tell Python
+  to :ref:`treat warnings as errors <python:warning-filter>` using, for
+  example, ``python -Werror manage.py shell`` which would give you a complete
+  stack trace.
```

Comment:
Rewording to group the "as errors" and "stack trace" concepts before giving an example.

```suggestion
  To make it easier to find the code that triggered this warning, you can make
  Python :ref:`treat warnings as errors <python:warning-filter>` to reveal the
  stack trace, for example with ``python -Werror manage.py shell``.
```